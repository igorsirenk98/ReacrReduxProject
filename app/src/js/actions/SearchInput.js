import {
    SEARCH_INPUT_CHANGE
} from '../constants/action-types';

export const searchInputChange = value => ({
    type: SEARCH_INPUT_CHANGE,
        payload: {
            searchInputValue: value
        }
});