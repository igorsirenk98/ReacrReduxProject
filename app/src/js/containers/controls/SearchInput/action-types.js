export const SEARCH_INPUT_CHANGE = 'SEARCH_INPUT_CHANGE';