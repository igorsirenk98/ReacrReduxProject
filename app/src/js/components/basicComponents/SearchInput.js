import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import InputBase from '@material-ui/core/InputBase';
import { makeStyles } from '@material-ui/core/styles';
import MenuIcon from '@material-ui/icons/Menu';
import SearchIcon from '@material-ui/icons/Search';
import Tooltip from '@material-ui/core/Tooltip';

import { fetchProductsBySearch } from '../../actions/ProductsBySearch';
import { searchInputChange } from '../../actions/SearchInput';

const mapStateToProps = state => ({
    searchInputValue: state.searchInputChange.searchInputValue
});

class SearchInput extends Component {
    handleChange = e => {
        const { value } = e.target;
        this.props.dispatch(searchInputChange(value));
    }

    render() {
        const { searchInputValue } = this.props;
        const pathname = searchInputValue ? `/products/search=${searchInputValue}` :
                                            '/products/top-products';

        return (
            <>
                <Tooltip title="Click to search" arrow>
                    <Link to={{
                        pathname
                    }}>
                        <IconButton aria-label="search">
                            <SearchIcon />
                        </IconButton>
                    </Link>
                </Tooltip>
                <InputBase
                    value={searchInputValue}
                    placeholder="Search bikesâ€¦"
                    inputProps={{ 'aria-label': 'search' }}
                    onChange={this.handleChange}
                />
            </> 
        )
    }
}

export default connect(mapStateToProps)(SearchInput);