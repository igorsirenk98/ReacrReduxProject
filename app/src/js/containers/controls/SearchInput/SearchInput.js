import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { history } from '../../../history/history';

import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import IconButton from '@material-ui/core/IconButton';
import InputBase from '@material-ui/core/InputBase';
import SearchIcon from '@material-ui/icons/Search';
import Tooltip from '@material-ui/core/Tooltip';

import { searchInputChange } from './actions';

const mapStateToProps = state => ({
    searchValue: state.searchInput.payload.searchValue
});

const mapDispatchToProps = dispatch => {
    return bindActionCreators({
        searchInputChange
    }, dispatch);
};

export class SearchInput extends Component {
    handleChange = e => {
        const { value } = e.target;
        this.props.searchInputChange(value);
    }

    handleSearchSubmit = e => {
        if (e.keyCode === 13) {
            history.push(this.generateUrl());
        }
    }

    generateUrl() {
        const { searchValue } = this.props;
        const pathname = searchValue ?
            `/products/search=${encodeURIComponent(searchValue)}` : '/';

        return pathname;
    }

    render() {
        const { searchValue } = this.props;

        return (
            <>
                <Tooltip title="Click to search" arrow>
                    <Link to={this.generateUrl()}>
                        <IconButton className="header__search-icon" aria-label="search">
                            <SearchIcon />
                        </IconButton>
                    </Link>
                </Tooltip>
                <InputBase
                    className="header__input"
                    value={searchValue}
                    placeholder="Search bikesâ€¦"
                    inputProps={{ 'aria-label': 'search' }}
                    onChange={this.handleChange}
                    onKeyDown={this.handleSearchSubmit}
                />
            </>
        )
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(SearchInput);