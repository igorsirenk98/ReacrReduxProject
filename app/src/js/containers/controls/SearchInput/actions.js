import { SEARCH_INPUT_CHANGE } from './action-types';

export const searchInputChange = value => ({
	type: SEARCH_INPUT_CHANGE,
	payload: {
		searchValue: value
	}
});
