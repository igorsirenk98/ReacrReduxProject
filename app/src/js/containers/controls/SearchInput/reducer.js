import { SEARCH_INPUT_CHANGE } from './action-types';

const initialState = {
	payload: {
		searchValue: ''
	}
};

const searchInput = (state = initialState, action) => {
	switch (action.type) {
		case SEARCH_INPUT_CHANGE: {
			return {
				...state,
				payload: action.payload
			};
		}
		default: {
			return state;
		}
	}
};

export default searchInput;
