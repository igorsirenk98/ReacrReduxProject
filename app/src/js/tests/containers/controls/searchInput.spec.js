import React from 'react';
import { shallow } from 'enzyme';
import { SearchInput } from '../../../containers/controls/SearchInput/SearchInput';

describe('SearchInput component', () => {
    const props = {
        searchValue: '',
        searchInputChange: () => { },
    }

    describe('SearchInput initial', () => {
        const searchInput = shallow(<SearchInput {...props}/>);

        it('renders correctly', () => {
            expect(searchInput.debug()).toMatchSnapshot()
        })
    });
    
    describe('SearchInput with entered value', () => {
        const nextProps = {
            ...props,
            searchValue: 'a',
        }
        const searchInput = shallow(<SearchInput {...nextProps}/>);

        it('renders correctly', () => {
            expect(searchInput.debug()).toMatchSnapshot()
        })
    });
})
