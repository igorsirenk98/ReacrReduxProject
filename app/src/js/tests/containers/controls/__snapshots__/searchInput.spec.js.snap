// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput component SearchInput initial renders correctly 1`] = `
"<Fragment>
  <WithStyles(ForwardRef(Tooltip)) title=\\"Click to search\\" arrow={true}>
    <Link to=\\"/\\">
      <WithStyles(ForwardRef(IconButton)) className=\\"header__search-icon\\" aria-label=\\"search\\">
        <SearchIcon />
      </WithStyles(ForwardRef(IconButton))>
    </Link>
  </WithStyles(ForwardRef(Tooltip))>
  <WithStyles(ForwardRef(InputBase)) className=\\"header__input\\" value=\\"\\" placeholder=\\"Search bikes…\\" inputProps={{...}} onChange={[Function]} onKeyDown={[Function]} />
</Fragment>"
`;

exports[`SearchInput component SearchInput with entered value renders correctly 1`] = `
"<Fragment>
  <WithStyles(ForwardRef(Tooltip)) title=\\"Click to search\\" arrow={true}>
    <Link to=\\"/products/search=a\\">
      <WithStyles(ForwardRef(IconButton)) className=\\"header__search-icon\\" aria-label=\\"search\\">
        <SearchIcon />
      </WithStyles(ForwardRef(IconButton))>
    </Link>
  </WithStyles(ForwardRef(Tooltip))>
  <WithStyles(ForwardRef(InputBase)) className=\\"header__input\\" value=\\"a\\" placeholder=\\"Search bikes…\\" inputProps={{...}} onChange={[Function]} onKeyDown={[Function]} />
</Fragment>"
`;
